### DISCUSSION TITLE
创建触发器成功但执行报错：NoClassDefFoundError

### DISCUSSION THREAD

**Discussion by StrongManLi** [apache/iotdb#14010] [General]
创建Trigger的语句是：
``` 
CREATE STATELESS TRIGGER triggerTest BEFORE INSERT ON root.test.** AS 'com.test.trigger.TestTrigger' USING URI "file:///IotDB/apache-iotdb-1.3.2-all-bin/apache-iotdb-1.3.2-all-bin/ext/trigger/TestTrigger-1.0-SNAPSHOT.jar" WITH (    "ip" = "127.0.0.1", "port" = "6667");
```
但是执行时报错 NoClassDefFoundError。找不到的类是一个在 com.test.trigger.iotdb 目录下的Util类，触发器是在com.test.trigger目录下。在打包好的Jar包中能够找到这个Util类对应的 class 文件。
创建的触发器为什么没有读取到这个Util类呢？是创建 Trigger的语句不对吗？ v1.3.2
![image](https://github.com/user-attachments/assets/e7c83426-ecd5-4a6d-a437-5cdb728c94ee)

**Comment by CritasWang** [comment]
触发器的包是 with-dependencies的吗

**Reply by StrongManLi** [reply]
不是，这个需要 with-dependencies 吗？

**Reply by StrongManLi** [reply]
之前测试用的一个Trigger也是这么打包的也能正常执行，只不过之前的那个就一个类，这个有多个，并没有引入额外的依赖。

**Comment by StrongManLi** [comment]
内部类也无法被读取到，只使用了一个类。然后还有一个问题就是：
Drop 一个 trigger 时 install 里的类没有删除，这个应该不是问题，但是我在重新上传同名 jar 包并创建一个同名的 trigger之后 tmp 目录里的内容还是指向原来的 jar，而且好像这些jar有缓存或者什么，重启iotdb会找回好几个已经删除了的jar在install目录里

**Reply by coniferous-cmd** [reply]
@StrongManLi  我也遇到了这个问题请问您解决了吗
